pkg/framework/deployment.go

```diff
@@ -66,6 +78,27 @@ func IsDeploymentAvailable(c client.Client, name, namespace string) bool {
 	return true
 }
 
+// IsDeploymentSynced returns true if the deployment has one or more availabe replicas
```

Comment:
i think the comment is a little off here, the function looks like it's checking equality between the deployment specs not just replicas.